src/datasets/load.py

```diff
@@ -1791,10 +1791,10 @@ def dataset_module_factory(
                     raise e
             if filename in [sibling.rfilename for sibling in dataset_info.siblings]:  # contains a dataset script
                 fs = HfFileSystem(endpoint=config.HF_ENDPOINT, token=download_config.token)
-                if _require_custom_configs:
+                if _require_custom_configs or (revision and revision != "main"):
                     can_load_config_from_parquet_export = False
                 elif _require_default_config_name:
-                    with fs.open(f"datasets/{path}/{filename}", "r", revision=revision, encoding="utf-8") as f:
+                    with fs.open(f"datasets/{path}/{filename}", "r", encoding="utf-8") as f:
```

Comment:
I have also removed passing `revision` to `fs.open`: 3acefe542dd4b1f06a879b6c1081089e95588f9d
- Previous condition makes this unnecessary
- Additionally, note that this has never worked (the revision is ignored). I have opened a PR to fix it: 
  - huggingface/huggingface_hub#1948

Reply:
good catch !